src/util/artifacts.js

```diff
@@ -18,8 +23,20 @@ export async function prepareArtifactsDir(
     }
   } catch (error) {
     if (isErrorWithCode('ENOENT', error)) {
-      log.debug(`Creating artifacts directory: ${artifactsDir}`);
-      await fs.mkdir(artifactsDir);
+      try {
```

Comment:
This is a great improvement on the error messages. Also do we really need `await mkdirp` outside of existing project conventions?